src/clj_jargon/permissions.clj

```diff
@@ -153,65 +152,93 @@
      :write write
      :own   own}))
 
-(defn dataobject-perm?
+(defn- user-id-dataobject-perm?
+  "Utility function which checks if a given user-id has a given permission on
+  data-path"
+  [cm user-id data-path perm]
+  (let [results (dataobject-perms-rs cm user-id data-path)]
+    (some #(get (perm-map-for (first (.getColumnsAsList %))) perm) results)))
+
+(defn- loop-user-id-dataobject-perms
+  "Iterates through user IDs, checking if that user has the specified
+  permission or better on data-path, short-circuiting when true"
+  [cm user-ids data-path perm]
+  (when (seq user-ids)
+    (or (user-id-dataobject-perm? cm (first user-ids) data-path perm)
+        (recur cm (rest user-ids) data-path perm))))
```

Comment:
Ah, it does look like you're right. I thought I'd tried using `some` but I must have done it some different way which resulted in evaluating everything first. But looking at [the implementation](https://github.com/clojure/clojure/blob/clojure-1.9.0-alpha14/src/clj/clojure/core.clj#L2665) it's clear that `some` basically does the same thing. I'll look into changing this, that's a lot cleaner than reimplementing basically exactly what `some` does, heh.